app/plugin.go

```diff
@@ -638,6 +625,83 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 	return nil
 }
 
+func (a *App) installPrepackagedPlugins(pluginsDir string) []*plugin.PrepackagedPlugin {
+	prepackagedPluginsDir, found := fileutils.FindDir(pluginsDir)
+	if !found {
+		return nil
+	}
+
+	fileStorePaths := []string{}
+	if err := filepath.Walk(prepackagedPluginsDir, func(walkPath string, info os.FileInfo, err error) error {
+		fileStorePaths = append(fileStorePaths, walkPath)
+		return nil
+	}); err != nil {
+		mlog.Error("Failed to walk prepackaged plugins", mlog.Err(err))
+	}
+	pluginSignaturePathMap := getPluginsFromFilePaths(fileStorePaths)
+	plugins := make([]*plugin.PrepackagedPlugin, 0, len(pluginSignaturePathMap))
+	for _, pluginPaths := range pluginSignaturePathMap {
+		plugin, err := a.installPrepackagedPlugin(pluginPaths)
+		if err != nil {
+			mlog.Error("Failed to install prepackaged plugin %s", mlog.Err(err), mlog.String("path", pluginPaths.path))
+			continue
+		}
+		plugins = append(plugins, plugin)
+	}
+	return plugins
+}
+
+func (a *App) installPrepackagedPlugin(pluginPaths *pluginSignaturePaths) (*plugin.PrepackagedPlugin, error) {
+	fileReader, err := os.Open(pluginPaths.path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to open prepackaged plugin %s", pluginPaths.path)
+	}
+	tmpDir, err := ioutil.TempDir("", "plugintmp")
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create temp dir plugintmp")
+	}
+	defer os.RemoveAll(tmpDir)
+
+	plug, pluginDir, err := getPrepackagedPlugin(pluginPaths, fileReader, tmpDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to get prepackaged plugin %s", pluginPaths.path)
+	}
+
+	if _, err := a.installExtractedPlugin(plug.Manifest, pluginDir, true); err != nil {
+		return nil, errors.Wrapf(err, "Failed to install extracted prepackaged plugin %s", pluginPaths.path)
+	}
+
+	return plug, nil
+}
+
+func getPrepackagedPlugin(pluginPaths *pluginSignaturePaths, pluginFile io.ReadSeeker, tmpDir string) (*plugin.PrepackagedPlugin, string, error) {
+	manifest, pluginDir, err := extractPlugin(pluginFile, tmpDir)
+	if err != nil {
+		return nil, pluginDir, errors.Wrapf(err, "Failed to extract plugin with path %s", pluginPaths.path)
+	}
+
+	plugin := new(plugin.PrepackagedPlugin)
+	plugin.Manifest = manifest
+	plugin.Signatures = make([][]byte, 0, len(pluginPaths.signaturePaths))
+
+	for _, sig := range pluginPaths.signaturePaths {
+		sigReader, sigErr := os.Open(sig)
+		if sigErr != nil {
+			return nil, pluginDir, errors.Wrapf(sigErr, "Failed to open prepackaged plugin signature %s", sig)
+		}
+		bytes, sigErr := ioutil.ReadAll(sigReader)
+		if sigErr != nil {
+			return nil, pluginDir, errors.Wrapf(sigErr, "Failed to read prepackaged plugin signature %s", sig)
+		}
+		plugin.Signatures = append(plugin.Signatures, bytes)
+	}
+	plugin.IconData, err = getIcon(manifest.IconPath)
+	if err != nil {
+		mlog.Error("Failed to get icon", mlog.Err(err), mlog.String("icon path", manifest.IconPath))
```

Comment:
If `getIcon` fails, there is likely a error in the release of the plugin. I'm 1/5 that we just return an error in this case for better visibility and not continue.

The error should happen quite rarely, but if it happens, it should be fixed ASAP from our side.